File: models/curso.py
from sqlalchemy import Column, Integer, String, Boolean, ForeignKey
from sqlalchemy.dialects.postgresql import JSONB
from config.db import Base

class Curso(Base):
    __tablename__ = "curso"

    id_curso = Column(Integer, primary_key=True, index=True)
    nombre = Column(String(120), nullable=False)
    cupo_estudiante = Column(Integer, nullable=False)
    horario = Column(JSONB, nullable=False)
    id_docente = Column(Integer, ForeignKey("docente.id_docente"))
    estado = Column(Boolean, default=True)

File: src/models/Curso.js
export class Curso {
  constructor({
    id_curso,
    nombre,
    cupo_estudiante,
    horario,
    id_docente,
    estado,
  }) {
    this.id_curso = id_curso;
    this.nombre = nombre;
    this.cupo_estudiante = cupo_estudiante;
    this.horario = horario; // JSONB en el backend
    this.id_docente = id_docente;
    this.estado = estado;
  }
}
